tools/dockerfile/interoptest/grpc_interop_go/Dockerfile

```diff
@@ -23,23 +23,34 @@ RUN ln -s /usr/local/go/bin/go /usr/local/bin
 # Basic python dependencies to be able to run tools/run_tests python scripts
 # These dependencies are not sufficient to build gRPC Python, gRPC Python
 # deps are defined elsewhere (e.g. python_deps.include)
+# This template is similar to run_tests_python_deps.include but installs the
+# dependencies in a python virtual environment. This is required from pip 23.0
+# onwards as installation of system-wide packages is blocked by default.
 RUN apt-get update && apt-get install -y \
   python3 \
   python3-pip \
   python3-setuptools \
   python3-yaml \
+  python3-venv \
   && apt-get clean
 
-# use pinned version of pip to avoid sudden breakages
-RUN python3 -m pip install --upgrade pip==19.3.1
+# Create a python virtual env for installing required dependencies.
+ENV PYTHON_ENV="/opt/venv/grpc-interop"
+RUN python3 -m venv "$PYTHON_ENV"
+RUN echo "source $PYTHON_ENV/bin/activate" >> ~/.bashrc
 
-# TODO(jtattermusch): currently six is needed for tools/run_tests scripts
+# Use bash explicitly as /bin/sh doesn't support the "source" command.
+# use pinned version of pip to avoid sudden breakages
+# TODO(#36814): currently six is needed for tools/run_tests scripts
 # but since our python2 usage is deprecated, we should get rid of it.
-RUN python3 -m pip install six==1.16.0
+RUN /bin/bash -c "source $PYTHON_ENV/bin/activate &&   python3 -m pip install --upgrade pip==19.3.1 &&   python3 -m pip install six==1.16.0"
 
-# Google Cloud Platform API libraries
+# Install Google Cloud Platform API libraries
 # These are needed for uploading test results to BigQuery (e.g. by tools/run_tests scripts)
-RUN python3 -m pip install --upgrade google-auth==1.23.0 google-api-python-client==1.12.8 oauth2client==4.1.0
+# This template is similar to gcp_api_libraries.include but installs the
+# dependencies in a python virtual environment. This is required from pip 23.0
+# onwards as installation of system-wide packages is blocked by default.
+RUN /bin/bash -c "source $PYTHON_ENV/bin/activate &&   python3 -m pip install --upgrade google-auth==1.23.0 google-api-python-client==1.12.8 oauth2client==4.1.0"
```

Comment:
These dont seem to match the template. Was this pending rebase of https://github.com/arjan-bal/grpc/commit/a535a11efc2385e852d368e7025a15f7c6615293

Reply:
Its looks the correct to me. The command on line 46 installs pip and six, then on line 53 we install google-auth, google-api-python-client and oauth2client.
Before https://github.com/arjan-bal/grpc/commit/a535a11efc2385e852d368e7025a15f7c6615293, google-auth, google-api-python-client and oauth2client were installed twice (on line 46 and line 53).

Reply:
I see. Thanks for clarifying.